### PR TITLE
fix: Accepted Media Types settings validation

### PR DESCRIPTION
**Pull request by hugocostadev**
## Proposed changes (including videos or screenshots)

The setting `Accepted Media Types` in the File Upload menu was not working as expected since an Empty setting or  wildcard `*` value should allow all types of files if they are not in the `Blocked` settings

## Issue(s)

fix https://github.com/RocketChat/Rocket.Chat/issues/31906
fix https://github.com/RocketChat/Rocket.Chat/issues/31692

## Steps to test or reproduce

1. Login to rocket chat.
2. Set "Accepted Media Types" setting in "File Upload"  with `*` or leave it empty
3. Try sending any file that is not in the Blocked List
4. The file is not uploaded

## Further comments

https://rocketchat.atlassian.net/browse/CONN-214

### CODE REVIEW

**Review by KevLehman — DISMISSED** [review]
This one should have API tests

**Comment by hugocostadev** [comment]
> This one should have API tests

I created tests for the function that I changed and that's being used in frontend and backend. 
I agree that would be good to have api tests for that endpoint, I can do it, but I have others priorities in the moment. So would take a while

**Review by rique223 — DISMISSED** [review]

File: apps/meteor/app/utils/lib/restrictions.ts [hunk 1]
```diff
@@ -17,37 +15,45 @@ const fileUploadMediaBlackList = function (customBlackList: string): string[] |
...
 	}
 
-	if (blackList && isTypeOnList(type, blackList)) {
+	if (whiteList && type && isTypeOnList(type, whiteList)) {
+		return true;
+	}
+
+	if (!type && whiteList) {
 		return false;
 	}
```
Comment (matheusbsilva137) [inline apps/meteor/app/utils/lib/restrictions.ts:52]:
```suggestion
	if (whiteList) {
		return isTypeOnList(type, whiteList);
	}
```
Is believe this works just the same, doesn't it? Looks easier to read

**Review by matheusbsilva137 — CHANGES_REQUESTED** [review]

File: apps/meteor/app/utils/lib/restrictions.ts (hunk 1 above)
Comment (hugocostadev) [reply apps/meteor/app/utils/lib/restrictions.ts]: It does, but I had to make some smalls adjustments to `isTypeOnList` func ... Thanks for your feedback

**Review by rique223 — APPROVED** [review]

**Review by matheusbsilva137 — APPROVED** [review]

**Pull request merged**